### ISSUE TITLE
github errors, but tagbot appears as successful

### ISSUE THREAD

**Issue by longemen3000** [JuliaRegistries/TagBot#238]
in the last days, i've been experimenting consistent tagbot failures on what it seems to be a server error? this is the stacktrace from the tagbot workflow:
```
/usr/bin/docker run --name ghcriojuliaregistriestagbot1141_f9a324 --label 08450d --workdir /github/workspace --rm -e INPUT_TOKEN -e INPUT_SSH -e INPUT_REGISTRY -e INPUT_GITHUB -e INPUT_GITHUB_API -e INPUT_BRANCHES -e INPUT_DISPATCH -e INPUT_DISPATCH_DELAY -e INPUT_LOOKBACK -e INPUT_SSH_PASSWORD -e INPUT_GPG -e INPUT_GPG_PASSWORD -e INPUT_REGISTRY_SSH -e INPUT_DRAFT -e INPUT_USER -e INPUT_EMAIL -e INPUT_BRANCH -e INPUT_CHANGELOG -e INPUT_CHANGELOG_IGNORE -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/Clapeyron.jl/Clapeyron.jl":"/github/workspace" ghcr.io/juliaregistries/tagbot:1.14.1
Warning: GitHub returned a 5xx error code
Traceback (most recent call last):
  File "/root/tagbot/action/__main__.py", line 74, in <module>
    versions = repo.new_versions()
  File "/root/tagbot/action/repo.py", line 432, in new_versions
    return self._filter_map_versions(versions)
  File "/root/tagbot/action/repo.py", line 271, in _filter_map_versions
    expected = self._commit_sha_from_registry_pr(version, tree)
  File "/root/tagbot/action/repo.py", line 205, in _commit_sha_from_registry_pr
    pr = self._registry_pr(version)
  File "/root/tagbot/action/repo.py", line 196, in _registry_pr
    for pr in prs:
  File "/usr/local/lib/python3.8/site-packages/github/PaginatedList.py", line 56, in __iter__
    newElements = self._grow()
  File "/usr/local/lib/python3.8/site-packages/github/PaginatedList.py", line 67, in _grow
    newElements = self._fetchNextPage()
  File "/usr/local/lib/python3.8/site-packages/github/PaginatedList.py", line 199, in _fetchNextPage
    headers, data = self.__requester.requestJsonAndCheck(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 315, in requestJsonAndCheck
    return self.__check(
  File "/usr/local/lib/python3.8/site-packages/github/Requester.py", line 340, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 502 {"message": "Server Error"}
```
I would expect that the TagBot action would fail, but appears as successful after that

**Comment by gdalle** [comment]
Hi! Are there any updates on this?

**Comment by longemen3000** [comment]
I had to tag the release manually. Also, the registration was done via juliahub

**Comment by epatters** [comment]
I am also experiencing this issue. I tried re-running the TagBot action a couple times, to no avail.

**Comment by mfiano** [comment]
I have been unable to get TagBot to work due to this error for a week of several PR merges into General, triggered from both JuliaHub and `@JuliaRegistrator register`. It seems TagBot is just broken for a select few repositories with no explanation as to why. (I checked other recent General merges that were all successfully tagged/released)

https://github.com/mfiano/CoherentNoise.jl/runs/7947518284?check_suite_focus=true#step:3:70

**Comment by tecosaur** [comment]
For what its worth, I seem to be running into this too (https://github.com/tecosaur/DataToolkit.jl/actions/runs/3433997912/jobs/5724898705).

**Comment by codetalker7** [comment]
Hi. I have been experiencing this kind of issue as well. Please see this run: https://github.com/xKDR/TSFrames.jl/actions/runs/3829930514/jobs/6517161243

**Comment by epatters** [comment]
In case it's helpful: I still experience this issue. As a workaround, I use the Docker container to run the action manually. Sometimes it takes dozens of manual runs before it successfully completes, but eventually it works. I have no idea why it is so unreliable.

**Comment by PharosAbad** [comment]
"I have no idea why it is so unreliable."  for recent 2 weeks, it always fire "Internal server error" or other exceptional errors, but show as successful
https://github.com/PharosAbad/LightenQP.jl/actions/runs/4031075246/jobs/6930153475#step:3:45

**Comment by ericphanson** [comment]
The original post is now a troubleshooting tip: https://github.com/JuliaRegistries/TagBot#the-workflow-shows-as-successful-even-though-errors-have-occurred. Some of the other issues may have been fixed, especially with #264. I will close this but please feel free to open new ones.
